src/main/java/org/reaktivity/nukleus/socks/internal/stream/AbstractStream.java

```diff
@@ -0,0 +1,369 @@
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.nukleus.socks.internal.stream;
+
+import static org.reaktivity.nukleus.buffer.BufferPool.NO_SLOT;
+
+import org.agrona.DirectBuffer;
+import org.agrona.MutableDirectBuffer;
+import org.reaktivity.nukleus.function.MessageConsumer;
+import org.reaktivity.nukleus.socks.internal.stream.types.FragmentedFlyweight;
+import org.reaktivity.nukleus.socks.internal.stream.types.FragmentedForwarderBufferData;
+import org.reaktivity.nukleus.socks.internal.stream.types.FragmentedForwarderBufferEmpty;
+import org.reaktivity.nukleus.socks.internal.stream.types.FragmentedForwarderComplete;
+import org.reaktivity.nukleus.socks.internal.stream.types.FragmentedForwarderPartial;
+import org.reaktivity.nukleus.socks.internal.stream.types.FragmentedHandler;
+import org.reaktivity.nukleus.socks.internal.stream.types.FragmentedSenderComplete;
+import org.reaktivity.nukleus.socks.internal.stream.types.FragmentedSenderPartial;
+import org.reaktivity.nukleus.socks.internal.types.Flyweight;
+import org.reaktivity.nukleus.socks.internal.types.OctetsFW;
+import org.reaktivity.nukleus.socks.internal.types.stream.AbortFW;
+import org.reaktivity.nukleus.socks.internal.types.stream.BeginFW;
+import org.reaktivity.nukleus.socks.internal.types.stream.DataFW;
+import org.reaktivity.nukleus.socks.internal.types.stream.EndFW;
+import org.reaktivity.nukleus.socks.internal.types.stream.ResetFW;
+import org.reaktivity.nukleus.socks.internal.types.stream.WindowFW;
+
+public abstract class AbstractStream
+{
+    protected final Context context;
+
+    protected int slotIndex = NO_SLOT;
+    protected int slotReadOffset;
+    protected int slotWriteOffset;
+
+    protected int attemptOffset;
+
+    public AbstractStream(Context context)
+    {
+        this.context = context;
+    }
+
+    protected abstract void handleStream(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length);
+
+    protected void doBegin(
+        MessageConsumer streamConsumer,
+        long streamId,
+        long sourceRef,
+        long correlationId)
+    {
+        BeginFW beginToAcceptReply = context.beginRW
+            .wrap(context.writeBuffer, 0, context.writeBuffer.capacity())
+            .streamId(streamId)
+            .source("socks")
+            .sourceRef(sourceRef)
+            .correlationId(correlationId)
+            .extension(e -> e.reset())
```

Comment:
Please remove this line, it is no longer necessary since extension will default to empty anyway